### PR TITLE
Enable deterministic support for oneDNN

### PR DESCRIPTION
**Pull request by weishi-deng**
This PR is a part of RFC https://github.com/pytorch/pytorch/issues/114848.
For the request for Torchbenchmark models, this PR enables the deterministic attribute for the oneDNN operators for XPU backends, like convolution, deconvolution and matmult.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

File: torch/backends/__init__.py [hunk 1]
```diff
@@ -64,6 +64,7 @@ def __getattr__(self, attr):
     mkldnn as mkldnn,
     mps as mps,
     nnpack as nnpack,
+    onednn as onednn,
```
Comment (EikanWang) [inline torch/backends/__init__.py:67]: I'm not sure if we need to add `onednn` or reuse `mkldnn`. @jgong5 , any thoughts?

**Comment by EikanWang** [comment]
@weishi-deng , please fix lint issue.

**Comment by gujinghui** [comment]
Will we have PRs to enable the TF32, BF32 in oneDNN? @EikanWang

File: torch/backends/__init__.py (hunk 1 above)
Comment (jgong5) [reply torch/backends/__init__.py:67]: Better to use mkldnn for consistency.

**Review by jgong5 — CHANGES_REQUESTED** [review]
Besides the naming (use mkldnn instead of onednn for consistency), do we need to update the documentation somewhere since it is about API update?

**Comment by EikanWang** [comment]
> Will we have PRs to enable the TF32, BF32 in oneDNN? @EikanWang

Yes, we will.

File: torch/backends/__init__.py (hunk 1 above)
Comment (EikanWang) [reply torch/backends/__init__.py:67]: @weishi-deng , please use `mkldnn` first. In the future, we can add `onednn` if we think it is the right time to do so then.

**Comment by EikanWang** [comment]
> > Will we have PRs to enable the TF32, BF32 in oneDNN? @EikanWang
> 
> Yes, we will.

@gujinghui , @weishi-deng , https://github.com/pytorch/pytorch/pull/127294/files#diff-40b3d6cc2026cae8f139b15c4b0b05fd2e69c2715be0ce15648de79b0e15a4ebR365 FYI

File: torch/backends/__init__.py (hunk 1 above)
Comment (weishi-deng) [reply torch/backends/__init__.py:67]: Yes, will do.

**Review by EikanWang — APPROVED** [review]

File: aten/src/ATen/Context.cpp [hunk 2]
```diff
@@ -56,12 +56,12 @@ void Context::setDeterministicCuDNN(bool b) {
   deterministic_cudnn = b;
 }
 
-bool Context::deterministicOneDNN() const {
-  return deterministic_onednn;
+bool Context::deterministicMklDNN() const {
```
[the remark this answers is not in the document]
Comment (weishi-deng) [reply aten/src/ATen/Context.cpp:59]: Yes. I will fix it.

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
@desertfire , we're trying to enable deterministic for XPU and enable it in dynamo benchmarks for accuracy tests.

**Review by jgong5 — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

**Review by gujinghui — APPROVED** [review]

**Pull request closed without merging**